src/neuroglancer/volume_rendering/volume_render_layer.ts

```diff
@@ -165,7 +185,42 @@ void emitTransparent() {
   emitRGBA(vec4(0.0, 0.0, 0.0, 0.0));
 }
 `);
-        builder.setFragmentMainFunction(`
+        if (wireFrame) {
+          builder.setFragmentMainFunction(`
+void main() {
+  outputColor = vec4(uChunkNumber, uChunkNumber, uChunkNumber, 1.0);
+  emit(outputColor, 0u);
+}
+`)
+        } else {
+          let glslSnippets: VolumeRenderingShaderSnippets;
+          switch (shaderParametersState.mode) {
+            case VOLUME_RENDERING_MODES.ON:
+              glslSnippets = {
+                intensityCalculation: `
+    userMain();
+`,
+                beforeColorEmission: ``
+              };
+              break;
+            case VOLUME_RENDERING_MODES.MAX:
+              glslSnippets = {
+                intensityCalculation: `
+    float normChunkValue = toNormalized(getInterpolatedDataValue(0));
```

Comment:
I guess this is fine for experimentation purposes, but it is a bit unfortunate to hardcode this here --- it would be better if the user can somehow determine the input to the `max` function.  For example if the input is inverted the user would need to be able to negate it first.